packages/@aws-cdk/aws-rds/lib/database-secret.ts

```diff
@@ -55,5 +66,20 @@ export class DatabaseSecret extends secretsmanager.Secret {
         excludeCharacters: props.excludeCharacters ?? DEFAULT_PASSWORD_EXCLUDE_CHARS,
       },
     });
+
+    if (props.overrideLogicalId) {
+      const hash = crypto.createHash('md5');
+      hash.update(JSON.stringify({
+        path: this.node.path, // make it unique
+        // Use here the options that influence the password generation.
+        // If at some point we add other password customization options
+        // they sould be added here below (e.g. `passwordLength`).
+        excludeCharacters: props.excludeCharacters,
+      }));
+      const logicalId = `${this.node.path.replace(/([^A-Za-z0-9]|\/)/g, '')}${hash.digest('hex')}`;
+
+      const secret = this.node.defaultChild as secretsmanager.CfnSecret;
+      secret.overrideLogicalId(logicalId.slice(0, 255));
```

Comment:
Hmm... I think you want to take the _last_ 255 characters, not the first ones, no? Otherwise, you risk losing the hash?

How about using the new `addr` method of `ConstructNode` instead? Should simplify things. Maybe with combination with `id`, to make it readable? (I'm not sure what `addr` returns, too new 😜).

Reply:
`addr` returns a 42 characters hexadecimal string. Lets do `${id}${hash}` and take the last 255 chars.